### PR TITLE
PAYARA-812 Payara Fang

### PR DESCRIPTION
**Pull request by Pandrex247**
Introduces a new system app that provides Jolokia-type functionality.

Configured with the `set-payara-fang-configuration` command, which takes the following parameters:
* `--target` - The target instance
* `--enabled` - Whether or not the app/service is enabled
* `--contextroot` - The context root to deploy the app to
* `--name` - The system-app name
* `--securityenabled` - Enables password authentication

By app deploys to the admin listener, so with default settings the app deploys to localhost:4848/fang.

Password authentication is done via the _file_ realm - the app checks for a user in the _fang_ group, creating a default one if it doesn't exist.

### CODE REVIEW

**Comment by michaelranaldo** [comment]
Jenkins test please

**Comment by mulderbaba** [comment]
jenkins test please

File: appserver/payara-appserver-modules/payara-fang/payara-fang-war/pom.xml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>fish.payara.appserver.payara-fang</groupId>
+        <artifactId>payara-fang-parent</artifactId>
+        <version>4.1.2.172-SNAPSHOT</version>
```
Comment (Cousjava) [inline appserver/payara-appserver-modules/payara-fang/payara-fang-war/pom.xml]: Should this be changed to 4.1.2.173-SNAPSHOT?

**Comment by Pandrex247** [comment]
Jenkins, do the thing!

File: appserver/payara-appserver-modules/payara-fang/payara-fang-war/src/main/java/fish/payara/monitoring/fang/FangState.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+public final class FangState {
+
+    // Keys
+    private static final String AGENT_NAME_KEY = "agentName";
+    private static final String AGENT_VERSION_KEY = "agentVersion";
+    private static final String AGENT_TYPE_KEY = "agentType";
+
+    // Values
+    private static final String AGENT_NAME = "Fang";
+    private static final String AGENT_VERSION = "0.0.1-Alpha";
```
Comment (Cousjava) [inline appserver/payara-appserver-modules/payara-fang/payara-fang-war/src/main/java/fish/payara/monitoring/fang/FangState.java]: As you updated the version number for Fang, this needs updating as well

File: appserver/pom.xml [hunk 3]
```diff
@@ -149,6 +149,7 @@
 	<javax.management.j2ee-api.version>1.1.1</javax.management.j2ee-api.version>
         <jboss.logging.version>3.3.0.Final</jboss.logging.version>
         <guava.version>19.0</guava.version>
+        <payara.fang.version>0.0.1-ALPHA</payara.fang.version>
```
Comment (Cousjava) [inline appserver/pom.xml]: This needs updating as well in order for it to build

**Comment by lprimak** [comment]
jenkins test please

**Comment by arjantijms** [comment]
Jenkins, do the thing!

**Comment by smillidge** [comment]
jenkins test please

**Comment by smillidge** [comment]
This failed to build

**Comment by lprimak** [comment]
I think this is due to  what @cousjava mentions in earlier comments

**Comment by Pandrex247** [comment]
Jenkins, would you kindly?

**Comment by Pandrex247** [comment]
Jenkins, do it! Do it now!

**Pull request merged**